From 0bea9d18277e3b381b0b32b76347afc5d6f221da Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Sat, 30 Mar 2019 10:03:51 +0300
Subject: [PATCH] Don't modify /etc/mtab from temporary namespaces

These records are never removed and just pollute
/etc/mtab if that's a regular file.
And if /etc/mtab isn't a regular file,
then attempts to modify it are pointless.
---
 lib/portage/process.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/portage/process.py b/lib/portage/process.py
index 0dba55de3..dfbda75de 100644
--- a/lib/portage/process.py
+++ b/lib/portage/process.py
@@ -617,7 +617,7 @@ def _exec(binary, mycommand, opt_name, fd_pipes,
 								os._exit(1)
 							# mount new /proc for our namespace
 							s = subprocess.Popen(['mount',
-								'-t', 'proc', 'proc', '/proc'])
+								'-n', '-t', 'proc', 'proc', '/proc'])
 							mount_ret = s.wait()
 							if mount_ret != 0:
 								writemsg("Unable to mount new /proc: %d\n" % (mount_ret,),
-- 
2.19.2

